ivy/functional/backends/numpy/general.py

```diff
@@ -265,11 +265,11 @@ def one_hot(indices, depth, *, device):
     return res.reshape(list(indices.shape) + [depth])
 
 
-def shape(x: np.ndarray, as_array: bool = False) -> Union[tuple, np.ndarray]:
+def shape(x: np.ndarray, as_array: bool = False) -> Union[ivy.Shape, ivy.Array]:
     if as_array:
-        return np.asarray(np.shape(x))
+        return ivy.array(x.shape)
```

Comment:
actually we should have `ivy.array(np.shape(x))` or `ivy.Array(np.shape(x))` if `np.shape` returns a native shape instance.